### PR TITLE
New C wrappers for real and imaginary part of complex numbers

### PR DESCRIPTION
**Pull request by rdbyk**
I have replaced these three wrappers

```
CWRAPPER_OUTPUT_TYPE complex_real_part(basic s, const basic com);
CWRAPPER_OUTPUT_TYPE complex_double_real_part(basic s, const basic com);
CWRAPPER_OUTPUT_TYPE complex_mpc_real_part(basic s, const basic com);
```

by just the following one

```
CWRAPPER_OUTPUT_TYPE complex_base_real_part(basic s, const basic com)
{
    CWRAPPER_BEGIN
    SYMENGINE_ASSERT(is_a<ComplexBase>(*(com->m)));
    s->m = (down_cast<const ComplexBase &>(*(com->m))).real_part();
    CWRAPPER_END
}
```

And I did analogue changes regarding the `imaginary_part` wrappers.

So all in all 6 wrappers have been replaced by just 2, without loosing any functionality.

Hope you like it!

### CODE REVIEW

**Comment by rdbyk** [comment]
It seems that many/all CI checks failing ... I have compiled and tested locally on Ubuntu ...
what's wrong? Thanks for your patience.

File: symengine/cwrapper.cpp [hunk 1]
```diff
@@ -329,30 +330,29 @@ int real_mpfr_is_zero(const basic s)
     return (int)((down_cast<const RealMPFR &>(*(s->m))).is_zero());
 }
 
-#endif // HAVE_SYMENGINE_MPFR
```
Comment (isuruf) [inline symengine/cwrapper.cpp]: This needs to be kept here.

File: symengine/cwrapper.cpp [hunk 2]
```diff
@@ -330,29 +331,30 @@ int real_mpfr_is_zero(const basic s)
...
 {
     CWRAPPER_BEGIN
-    SYMENGINE_ASSERT(is_a<ComplexMPC>(*(com->m)));
-    s->m = (down_cast<const ComplexMPC &>(*(com->m))).imaginary_part();
+    SYMENGINE_ASSERT(is_a<ComplexBase>(*(com->m)));
+    s->m = (down_cast<const ComplexBase &>(*(com->m))).imaginary_part();
     CWRAPPER_END
 }
 
+#endif // HAVE_SYMENGINE_MPFR
```
Comment (isuruf) [inline symengine/cwrapper.cpp]: I think this is not necessary anymore.

File: symengine/cwrapper.cpp [hunk 3]
```diff
@@ -330,29 +331,29 @@ int real_mpfr_is_zero(const basic s)
...
-}
-
-CWRAPPER_OUTPUT_TYPE complex_mpc_real_part(basic s, const basic com)
+        
+CWRAPPER_OUTPUT_TYPE complex_base_real_part(basic s, const basic com)
 {
     CWRAPPER_BEGIN
-    SYMENGINE_ASSERT(is_a<ComplexMPC>(*(com->m)));
-    s->m = (down_cast<const ComplexMPC &>(*(com->m))).real_part();
+    SYMENGINE_ASSERT(is_a<ComplexBase>(*(com->m)));
```
Comment (isuruf) [inline symengine/cwrapper.cpp]: `is_a<ComplexBase>` checks exact type. Use `is_a_Complex` for checking that it is complex.

Comment (rdbyk) [inline symengine/cwrapper.cpp]: Thanks, I will `SymEngine::is_a_Complex`.

**Comment by ShikharJ** [comment]
@rdbyk Just a suggestion. You can install `clang-format` and then run `./bin/test_format_local.sh` before committing your code. That will automatically take care of all the formatting issues.

**Comment by rdbyk** [comment]
@ShikharJ Thank you very much for the tip about clang-format.

**Comment by isuruf** [comment]
Thanks for the PR

**Pull request merged**